common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -446,6 +446,10 @@ def copy_from_published(root_block_id):
                 for child_block_id in block.fields.get('children', []):
                     copy_from_published(child_block_id)
 
+                # If block is not the main container being discarded.
+                if root_block_id != BlockKey.from_usage_key(location):
+                    block_key = location.course_key.make_usage_key(root_block_id.type, root_block_id.id)
+                    self.remove_reference_if_moved(block_key, location, user_id)
```

Comment:
Is this code intentionally placed after the recursive call so that all the `_update_block_in_structure` calls are made before the `remove_reference_if_moved` calls? Or not? Are there any implications to this ordering?